pytools/graph.py

```diff
@@ -212,10 +212,10 @@ class CycleError(Exception):
     """
     Raised when a topological ordering cannot be computed due to a cycle.
 
-    :attr node: Node in a directed graph that is part of a cycle.
+    :attr nodes: List of nodes in a directed graph that are part of a cycle.
     """
-    def __init__(self, node: T) -> None:
-        self.node = node
+    def __init__(self, nodes: List[T]) -> None:
+        self.nodes = nodes
```

Comment:
It might make sense to override the `repr` to only show at most 10 or so nodes, to avoid massive output.